site/app/controllers/admin/ReportController.php

```diff
@@ -601,6 +601,41 @@ private function getLateStatusMessage(LateDayInfo $ldi) {
         }
     }
 
+
+
+    #[Route("/courses/{_semester}/{_course}/reports/rainbow_grades_customization_save", methods: ["POST"])]
+    public function writetocustomization(): JsonResponse {
+        if (!empty($_SERVER['HTTP_X_REQUESTED_WITH']) && strtolower($_SERVER['HTTP_X_REQUESTED_WITH']) == 'xmlhttprequest') {
```

Comment:
Why do you need to access the `$_SERVER` global here?

Reply:
needed during development stage, no longer need it good point.